pyron-engine/src/main/scala/com/cloudentity/pyron/api/ApiHandler.scala

```diff
@@ -229,18 +242,21 @@ object ApiRequestHandler {
       ApiGroupMatcher.makeMatch(Option(vertxRequest.host()), Option(vertxRequest.path()).getOrElse(""), group.matchCriteria)
     }
 
+
   def findMatchingRule(apiGroup: ApiGroup, vertxRequest: HttpServerRequest): Option[RuleWithPathParams] = {
-    @tailrec def rec(basePath: BasePath, left: List[Rule]): Option[RuleWithPathParams] =
-      left match {
-        case rule :: tail =>
-          RuleMatcher.makeMatch(vertxRequest.method(), Option(vertxRequest.path()).getOrElse(""), basePath, rule.conf.criteria) match {
-            case Match(pathParams) => Some(RuleWithPathParams(rule, pathParams))
-            case NoMatch => rec(basePath, tail)
-          }
-        case Nil => None
-      }
+    @tailrec
+    def loop(basePath: BasePath, rules: List[Rule]): Option[RuleWithPathParams] = rules match {
```

Comment:
It reduces indentation, but I get your point. I think I did it in few other places. Sometimes method signature is short and it looks good especially if the only thing done is a 'match' or 'for' -> 'yield'. But here, signature is quite long, it would be better to get it on new line. Will do.